packages/svelte/src/compiler/phases/2-analyze/visitors/Attribute.js

```diff
@@ -38,6 +38,19 @@ export function Attribute(node, context) {
 		mark_subtree_dynamic(context.path);
 	}
 
+	// class={[...]} or class={{...}} or `class={x}` need clsx to resolve the classes
+	if (
+		node.name === 'class' &&
+		!Array.isArray(node.value) &&
+		node.value !== true &&
+		node.value.expression.type !== 'Literal' &&
+		node.value.expression.type !== 'TemplateLiteral' &&
+		node.value.expression.type !== 'BinaryExpression'
+	) {
+		mark_subtree_dynamic(context.path);
+		node.metadata.needs_clsx = true;
```

Comment:
would be cool if we could still detect which classnames are possible, using something akin to `gather_possible_values`, [otherwise this is a regression](https://svelte.dev/playground/hello-world?version=pr-14714#H4sIAAAAAAAAE22PwQrCMAyGX6XkooJ09zkFb77D5mFbowxiO9psKqXvbooKKt7Cly9_kgi2vSCUcEAip67Ok1FLNAOjWcEaTgNhgLKOwPcxexkIf03tx1GHGYkz69qA_3jvLKNliYHKDLPqqQ1hG-vFybnFMe201lUhjV1jP_tRda0vFfsJVfqyxAt8J8wla0lRMVcN946cjHg0mwxSVsWQnB_j7BHtlzPZKaD50Tqa8G1VxXOnPMR4YyjzYemYHg_KhPZCAQAA) from the `class:` directive. I don't think it needs to be super sophisticated in order to catch most common cases — basically things like `class={{ foo }}` should get the same pruning treatment as `class:foo` does today